packages/hawtio/src/plugins/camel/profile/Profile.tsx

```diff
@@ -0,0 +1,104 @@
+import { Card, CardBody, CardHeader, CardTitle, Skeleton, Text } from '@patternfly/react-core'
+import React, { useContext, useEffect, useState } from 'react'
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
+import { CamelContext } from '../context'
+import { ProfileData, profileService } from './profile-service'
+import { IResponse } from 'jolokia.js'
+import { log } from '../globals'
+
+export const Profile: React.FunctionComponent = () => {
+  const { selectedNode } = useContext(CamelContext)
+  const [isReading, setIsReading] = useState(true)
+  const [profileData, setProfileData] = useState<ProfileData[]>([])
+
+  useEffect(() => {
+    // Unregister old handles
+    profileService.unregisterAll()
```

Comment:
Instead of doing it at the beginning of `useEffect`, isn't it better to do it at the bottom as the returned function?  Note `useEffect` provides the cleanup step by returning a function at the bottom.